lib/nvme/nvme.c

```diff
@@ -248,6 +248,9 @@ nvme_probe_one(enum spdk_nvme_transport transport, spdk_nvme_probe_cb probe_cb,
 		ctrlr->probe_info = *probe_info;
 
 		TAILQ_INSERT_TAIL(&g_spdk_nvme_driver->init_ctrlrs, ctrlr, tailq);
+	} else {
+		/* The device is not supported */
+		return 1;
```

Comment:
This part looks correct (the code has moved to nvme_ctrlr.c, but the logic should be the same).